### PR TITLE
Issue 391: Update Pravega library versions and deployment script

### LINKED ISSUE #391: Update Pravega library versions and deployment script
**Problem description**
The Pravega driver deployment script is outdated and does not work out-of-the-box as of now. We need to update the necessary scripts and version to make it work.

### PR DESCRIPTION
**Pull request by RaulGracia**
**Change log description**
Updates Pravega libraries and deployment scripts.

**Purpose of the change**
Fixes #391.

**What the code does**
- Updates Pravega and Bookkeeper versions at the server side when deploying, as well as the Pravega client library for the benchmark.
- Updates the AMI used and the default types of images.
- Updates the deployment script to adapt to few changes: i) docker installation, ii) `cloudalchemy.node_exporter` naming, iii) device names, iv) Bookkeeper initialization command.
- Updated the README to reflect previous changes.
- Set scope `provided` to `driver-nats-streaming`. The reason for doing that is that `driver-nats-streaming` is transitively importing a very old version of Google Protobuf (`3.9.1` - Aug 06, 2019). Pravega also uses Google Protobuf. When upgrading Pravega, we found a `NoSuchMethodError` related to Google Protobuf and the reason was that the newer Pravega version are compiled with a more recent Google Protobuf version with incompatible changes (since version `3.17.2`). Having both Google Protobuf versions in the classpath leads to runtime errors, so I decided to prevent the oldest Google Protobuf version to be transitively imported by `driver-nats-streaming`. Given that NATS Streaming has been deprecated, I feel that it is a safe action to do (which can precede the removal of NATS Streaming until NATS JetStream is available).

**How to verify it**
- Build is passing.
- I can run the benchmark locally against Pravega standalone.
- I have deployed Pravega on AWS and executed the benchmark successfully.

### CODE REVIEW

**Comment by RaulGracia** [comment]
@merlimat please, have a look when you have the time. Thanks!

File: benchmark-framework/pom.xml [hunk 1]
```diff
@@ -145,6 +140,12 @@
             <artifactId>HdrHistogram</artifactId>
             <version>2.1.12</version>
         </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>driver-nats-streaming</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (merlimat) [inline benchmark-framework/pom.xml:148]: Was this change intended?

Comment (RaulGracia) [reply benchmark-framework/pom.xml:148]: Hi @merlimat, yes this change is intended. I tried to explain the justification in the PR description:
`Set scope provided to driver-nats-streaming. The reason for doing that is that driver-nats-streaming is transitively importing a very old version of Google Protobuf (3.9.1 - Aug 06, 2019). Pravega also uses Google Protobuf. When upgrading Pravega, we found a NoSuchMethodError related to Google Protobuf and the reason was that the newer Pravega version are compiled with a more recent Google Protobuf version with incompatible changes (since version 3.17.2). Having both Google Protobuf versions in the classpath leads to runtime errors, so I decided to prevent the oldest Google Protobuf version to be transitively imported by driver-nats-streaming. Given that NATS Streaming has been deprecated, I feel that it is a safe action to do (which can precede the removal of NATS Streaming until NATS JetStream is available).`

**Review by merlimat — APPROVED** [review]
